src/DotNetty.Codecs.Protobuf/ProtobufDecoder.cs

```diff
@@ -22,6 +22,8 @@ public ProtobufDecoder(MessageParser messageParser)
             this.messageParser = messageParser;
         }
 
+        public override bool IsSharable => true;
```

Comment:
The decoder depends on the MessageParser which is typically a specific POCO from proto gen, for instance:
https://github.com/Azure/DotNetty/blob/dev/test/DotNetty.Codecs.Protobuf.Tests/RoundTripTests.cs#L196
In another words, the parser is bound to a message type, this should not be shareable.

Reply:
Well, in a specific pipeline, it is shareable since it means that bytes mapped to the protobuf message type message.

Reply:
I also removed 'sealed' from the Encoders per https://github.com/Azure/DotNetty/issues/196